web/server/vue-cli/src/mixins/analysis-info-handling.mixin.js

```diff
@@ -0,0 +1,268 @@
+        if (!checkersGrouped[analyzer][groupForChecker]) {
+          checkersGrouped[analyzer][groupForChecker] = {};
+        }
+
+        checkersGrouped[analyzer][groupForChecker][checkerName] = checkerInfo;
+      }
+    }
+
+    this.checkersGroupedAndSorted = Object.fromEntries(this.analyzers.map(
+      analyzer => [ analyzer,
+        Object.fromEntries(
+          Object.keys(checkersGrouped[analyzer]).sort().map(
+            group => [ group,
+              Object.keys(checkersGrouped[analyzer][group]).sort().map(
+                checkerName => [ checkerName,
+                  checkersGrouped[analyzer][group][checkerName]
+                ])
+            ])
+        )
+      ]));
+
+    this.checkerGroupCounts = Object.fromEntries(this.analyzers.map(
+      analyzer => [ analyzer,
+        Object.fromEntries(
+          Object.keys(this.checkersGroupedAndSorted[analyzer]).map(
+            group => [ group, [
+              // [CountKeys.Enabled]:
+              Object.keys(this.checkersGroupedAndSorted[analyzer][group])
+                .map(checker =>
+                  this.checkersGroupedAndSorted[analyzer][group][checker][1]
+                    .enabled ? 1 : 0)
+                .reduce((a, b) => a + b, 0),
+              /* [CountKeys.Disabled]: */ -1, // Will be updated later!
+              // [CountKeys.Total]:
+              Object.keys(
+                this.checkersGroupedAndSorted[analyzer][group]
+              ).length
+            ]
+            ]))
+      ]));
+    Object.keys(this.checkerGroupCounts).map(analyzer =>
+      Object.keys(this.checkerGroupCounts[analyzer]).map(group => {
+        this.checkerGroupCounts[analyzer][group][CountKeys.Disabled] =
+          this.checkerGroupCounts[analyzer][group][CountKeys.Total] -
+          this.checkerGroupCounts[analyzer][group][CountKeys.Enabled];
+      }));
+    Object.keys(this.checkerGroupCounts).map(analyzer => {
+      const sum = Object.values(this.checkerGroupCounts[analyzer])
+        .reduce((as, bs) => [
+          as[CountKeys.Enabled]  + bs[CountKeys.Enabled],
+          as[CountKeys.Disabled] + bs[CountKeys.Disabled],
+          as[CountKeys.Total]    + bs[CountKeys.Total]
+        ]);
+      this.checkerGroupCounts[analyzer][GroupKeys.AnalyzerTotal] = sum;
+    });
+
+    return this.checkersGroupedAndSorted;
+  }
+}
+
+const AnalysisInfoHandlingMixin = {
+  mixins: [ VersionMixin ],
+  props: {
+    runId: { type: Object, default: () => null },
+    runHistoryId: { type: Object, default: () => null },
+    reportId: { type: Object, default: () => null },
+  },
+
+  methods: {
+    loadAnalysisInfo() {
+      const analysisInfoFilter = new AnalysisInfoFilter({
+        // Query a run's analysis info only if a run history ID is not explicit.
+        runId: (!this.runHistoryId ? this.runId : null),
+
+        runHistoryId: this.runHistoryId,
+        reportId: this.reportId
+      });
```

Comment:
As we discussed, it would be better to have parameter list for loadAnalysisInfo method that contains runId, runHistoryId, reportId instead of using the props. You have already called in this way the loadAnalysisInfo() in the AnalysisInfoDialog.vue component.

Reply:
Blimey... I think the original attempt at the refactoring perhaps was a bit *too* eager then...